### PR TITLE
Refactor SalesEngine for add'l repositories

### LINKED ISSUE #59: Refactor SalesEngine
## What

Refactor SalesEngine class to take in a hash of data files and output a suite of repositories. 

## How

We will need to pull code out of this class to create the FileIO class and also to allow each repo to take responsibility for its own object creation.

### LINKED ISSUE #96: Refactor SalesEngine to include all repositories
Need all repositories to be 'plugged in' to the SalesEngine.

### PR DESCRIPTION
**Pull request by mkrumholz**
### Changes

- Updated SalesEngine class to include TransactionRepo and InvoiceRepo

### Requested review on...

- Not completed yet: CustomerRepo, InvoiceItemRepo need to be included in here as well. 

### Checklist
- [x] Tests pass
- [x] Ready to be merged?

closes #96

### CODE REVIEW

File: lib/invoice_repository.rb [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+  def inspect
+    "#<#{self.class} #{@invoices.size} rows>"
+  end
+
   def initialize(filename)
     @invoices = create_invoices(filename)
   end
 
-  def create_invoies(filename)
+  def create_invoices(filename)
```
Comment (mkrumholz) [inline lib/invoice_repository.rb:12]: Yay for fixing stupid spelling errors 😂

File: lib/sales_engine.rb [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
+require_relative './invoice_repository'
 require_relative './item'
 require_relative './item_repository'
 require_relative './merchant'
 require_relative './merchant_repository'
 require_relative './sales_analyst'
+require_relative './transaction'
+require_relative './transaction_repository'
+require_relative './invoice_item'
+require_relative './invoice_item_repository'
```
Comment (mkrumholz) [inline lib/sales_engine.rb]: @NoahZinter do we need to require the customer and customer repo files here?

File: spec/sales_engine_spec.rb [hunk 3]
```diff
@@ -1,22 +1,57 @@
...
 require './data/merchant_mocks'
+require './data/Mockable'
 require './lib/file_io'
+require './lib/invoice_repository'
 require './lib/item_repository'
 require './lib/merchant_repository'
 require './lib/sales_engine'
 require './lib/sales_analyst'
+require './lib/transaction_repository'
+require './lib/invoice_item_repository'
```
Comment (mkrumholz) [inline spec/sales_engine_spec.rb]: Here too -- @NoahZinter do we need to require customer and customer repo files here?

**Comment by mkrumholz** [comment]
I just resolved the merge conflicts with the Invoices one in my local, too -- will do that again before we merge there, but that shouldn't be an issue anymore.

**Review by redferret — APPROVED** [review]

File: lib/sales_analyst.rb [hunk 4]
```diff
@@ -60,23 +60,23 @@ def average_item_price_for_merchant(merchant_id)
 
   def average_average_price_per_merchant
     sum_of_averages = all_merchants.sum do |merchant|
-      average_item_price_for_merchant(merchant.id) unless num_of_items_per_merchant[merchant] == 0
+      average_item_price_for_merchant(merchant.id) unless num_of_items_per_merchant[merchant].zero?
```
Comment (sami-p) [inline lib/sales_analyst.rb:63]: Like this refactor 👍🏼

**Review by sami-p — APPROVED** [review]
Looks great! Merging now - access to move forward.

**Pull request merged**